dask/utils.py

```diff
@@ -1064,7 +1064,6 @@ def is_arraylike(x):
     return (
         hasattr(x, "shape")
         and isinstance(x.shape, tuple)
-        and x.shape
```

Comment:
@mrocklin I know we had [this discussion before](https://github.com/dask/dask/pull/4954#discussion_r295854570). But for this is what happens:

```python
>>> True and ()
()
```

So in https://github.com/dask/dask/blob/master/dask/array/utils.py#L111, a Dask array with shape `()` will never resolve to a NumPy array.

I don't know of a cleaner way to do fix this here, but feel free to suggest a more appropriate one.

Reply:
I don't understand the broader issue, but on the narrow issue of `True and ()` would wrapping this whole `return ( ... )` in a `return bool( ... )` fix it?

Reply:
Unfortunately not. The thing is that `bool(True and ())` will return `False`. It seems like this works well on the Dataframe side, but we have some Array `meta` where `shape=()` and they are `array_like`, so that entire expression must evaluate to `True`